### PR TITLE
Move seed function to custom Lambda CF resource, and some other small changes

### PR DESCRIPTION
**Pull request by matwerber1**
Change summary: 
#############

1. Move seed function from local script to custom CFN resource

2. Make userArn (for Cloud9) optional parameter; if not specified, only the creator of the CDK stack itself will have Cloud9 access.

3. Add new optional vpcId parameter for Cloud9 / RDS resources; if not specified, default VPC will be used. 

4. Revert customized deployment commands for `npm run build` / `cdk deploy`, and related config files (e.g. `tsconfig.json`) to the standard CDK boilerplate. I don't think those modifications were adding value and they led to the need to modify things like `build.sh` for new non-ts files I was adding when normally this wouldn't be an issue. 

5. Rename DynamoDB table to XXX-V1, with idea being we can add additional tables in future to show different approaches. 

6. Modified DynamDB seed methodology to parse a CSV (easier to use than typing out JSON object)

### CODE REVIEW

File: lib/constructs/TemplateConstruct.ts [hunk 1]
```diff
@@ -99,7 +134,7 @@ export default class Template extends Construct {
     });
     new CfnOutput(this, "SqlDatabaseSecretArn", {
       exportName: `SqlDatabaseSecretArn`,
-      value: sqlDatabase.secret ? sqlDatabase.secret.secretArn : ""
+      value: sqlDatabase.secret!.secretArn
```
Comment (shafkevi) [inline lib/constructs/TemplateConstruct.ts:135]: Didn't know about this syntax, neat.

File: lib/constructs/TemplateConstruct.ts [hunk 2]
```diff
@@ -1,61 +1,66 @@
...
+        tableName: `${id}-v1`,
         billingMode: BillingMode.PAY_PER_REQUEST,
         removalPolicy: RemovalPolicy.DESTROY,
         partitionKey: {
-          name: "partitionKey",
-          type: AttributeType.STRING
-        },
-        sortKey: {
-          name: "sortKey",
+          name: "id",
```
Comment (shafkevi) [inline lib/constructs/TemplateConstruct.ts:60]: Why did you change this to id? I personally think that the primary keys of the table should have 100% generic names, so you can really use the table flexibly and follow the ol' 1 table design.

File: lib/stacks/TemplateStack.ts [hunk 3]
```diff
@@ -4,6 +4,9 @@ import TemplateConstruct from "../constructs/TemplateConstruct";
 export default class Template extends Stack {
   constructor(scope: Construct, id: string, props?: StackProps) {
     super(scope, id, props);
-    new TemplateConstruct(this, id, { });
+    new TemplateConstruct(this, id, { 
+      vpcId: 'vpc-0a2cad50c98aed83f'    // optional param to use existing VPC; if not specified, will use default VPC.
```
Comment (shafkevi) [inline lib/stacks/TemplateStack.ts]: Probably shouldn't be hardcoding a vpc here.

**Review by shafkevi — APPROVED** [review]
There are a few smol pieces of feedback that I think should be addressed. But none of them are truly breaking issues, so I'll approve!

File: lib/constructs/TemplateConstruct.ts (hunk 2 above)
Comment (matwerber1) [reply lib/constructs/TemplateConstruct.ts:60]: totally agreed. I wanted to start with a DDB table that follows same mental model as relational, with the idea being that we'll add in additional DDB table(s) that show the evolution to a DDB best practices.

File: lib/stacks/TemplateStack.ts (hunk 3 above)
Comment (matwerber1) [reply lib/stacks/TemplateStack.ts]: Resolved - put in generic "vpc-xxxx" and commented out the line in commit 10c7aaa.
